src/graphql/node/control.rs

```diff
@@ -39,172 +38,231 @@ impl NodeControlMutation {
+    pub giganto_draft: Option<String>,
 }
 
-async fn send_set_config_requests(
-    agents: &BoxedAgentManager,
-    node: &Node,
-) -> anyhow::Result<Vec<ModuleName>> {
-    let profile_draft = node
-        .profile_draft
-        .as_ref()
-        .ok_or_else(|| anyhow::anyhow!("There is nothing to be applied."))?;
-
-    let mut result_combined: Vec<ModuleName> = vec![];
-
-    for (module_name, config) in target_app_configs(profile_draft) {
-        if send_set_config_request(
-            agents,
-            &profile_draft.hostname,
-            module_name.as_ref(),
-            &config,
-        )
-        .await?
-        {
-            result_combined.push(module_name);
-        }
-    }
+async fn notify_agents(
+    agent_manager: &BoxedAgentManager,
+    update_agent_keys: &[&str],
+    disable_agent_keys: &[&str],
+) -> Result<()> {
+    let update_futures = update_agent_keys.iter().map(|agent_key| async move {
+        agent_manager.update_config(agent_key).await.map_err(|e| {
+            async_graphql::Error::new(format!(
+                "Failed to notify agent for config update {agent_key}: {e}"
+            ))
+        })
+    });
 
-    Ok(result_combined)
-}
+    // TODO: We need to implement the logic to disable agents. For now, we will only log the message.
+    info!("Agents {disable_agent_keys:?} need to be notified to be disabled, but disabling logic is not yet implemented");
 
-async fn send_set_config_request(
-    agents: &BoxedAgentManager,
-    hostname: &str,
-    module_name: &str,
-    config: &review_protocol::types::Config,
-) -> anyhow::Result<bool> {
-    for _ in 0..MAX_SET_CONFIG_TRY_COUNT {
-        let set_config_response = agents.set_config(hostname, module_name, config).await;
-        if set_config_response.is_ok() {
-            return Ok(true);
-        }
-        info!("Failed to set config for module {module_name}. Retrying...");
-    }
+    let notification_results: Vec<Result<_>> = join_all(update_futures).await;
+
+    let errors: Vec<String> = notification_results
+        .into_iter()
+        .filter_map(|result| result.err().map(|e| e.message))
+        .collect();
 
-    Ok(false)
+    if errors.is_empty() {
+        Ok(())
+    } else {
+        Err(async_graphql::Error::new(errors.join("\n")))
+    }
 }
 
-fn target_app_configs(
-    _profile_draft: &NodeProfile,
-) -> Vec<(ModuleName, review_protocol::types::Config)> {
-    Vec::new() // TODO
+fn valid_node_apply(node: &Node) -> bool {
+    // check validity of node itself for apply
+    let valid_name_draft = node.name_draft.is_some();
+    let valid_profile_draft = node.profile_draft.is_some();
+
+    valid_name_draft && valid_profile_draft
 }
 
-#[allow(clippy::struct_excessive_bools)]
-struct ModuleSpecificProfileUpdateIndicator {
-    hog: bool,
-    reconverge: bool,
-    piglet: bool,
+// The scope of the applying node operation for a node.
+#[allow(clippy::enum_variant_names)]
+enum NodeApplyScope<'a> {
```

Comment:
I was wondering why all the variants include Giganto.

Reply:
I mean there might be a case where the _apply_ doesn't affect Giganto.

Reply:
How about handling the three steps—name and profile, agents, and Giganto—in sequence, instead of separating the cases?

Reply:
To me, `enum` variants seem to represent mutually exclusive cases, which is not the case.

Reply:
I made code changes on this point. I'd appreciate if you could check.